finch.yaml

```diff
@@ -148,27 +148,26 @@ provision:
 - mode: user
   script: |
     #!/bin/bash
+    # We need create /etc/nerdctl directory as root and make it writeable by non root user because the sftp client in https://github.com/runfinch/finch/blob/main/pkg/config/nerdctl_config_applier.go#L140 cannot run as root.
+    sudo mkdir -p /etc/nerdctl
+    sudo chown $USER /etc/nerdctl
+    sudo chmod 755 /root/
+    sudo chown $USER /root/.bashrc
     sudo chown $USER /mnt/lima-finch
 
-    # https://github.com/containerd/nerdctl/blob/cffdf87ff4d648a5344eea1406bb95ca3ad7eaa4/extras/rootless/containerd-rootless.sh#L144-L146
-    # XDG_DATA_HOME & ~/.local/share: https://github.com/containerd/nerdctl/blob/cffdf87ff4d648a5344eea1406bb95ca3ad7eaa4/extras/rootless/containerd-rootless.sh#L51
-    mkdir -p /mnt/lima-finch/containerd ~/.local/share/containerd
-    sudo mount --bind /mnt/lima-finch/containerd ~/.local/share/containerd
-
     # https://github.com/containerd/nerdctl/blob/main/docs/dir.md#dataroot
-    mkdir -p /mnt/lima-finch/nerdctl ~/.local/share/nerdctl
-    sudo mount --bind /mnt/lima-finch/nerdctl ~/.local/share/nerdctl
+    sudo mkdir -p /mnt/lima-finch/nerdctl /var/lib/nerdctl
+    sudo mount --bind /mnt/lima-finch/nerdctl /var/lib/nerdctl
 
     # https://github.com/containerd/nerdctl/blob/main/docs/dir.md#netconfpath
-    mkdir -p /mnt/lima-finch/cni-config ~/.config/cni
-    sudo mount --bind /mnt/lima-finch/cni-config ~/.config/cni
+    sudo mkdir -p /mnt/lima-finch/cni-config /etc/cni/
+    sudo mount --bind /mnt/lima-finch/cni-config /etc/cni/
+
+    sudo mkdir -p /var/lib/containerd
+    sudo mount --bind /mnt/lima-finch /var/lib/containerd
```

Comment:
Added reference doc. `/etc/containerd` has containerd related config files, in finch's case we have  [/etc/containerd/config.toml](https://github.com/containerd/containerd/blob/main/docs/man/containerd-config.toml.5.md#synopsis)  which is generated when the vm is initialized, we can't change this state unless the user logs into the vm and makes changes to this config. I am not sure if we want to save this data.

Reply:
> Could you do some manual verification for the test cases not yet covered in [additional_disk_test.go](https://github.com/runfinch/finch/blob/main/e2e/vm/additional_disk_test.go) and paste the test result into PR description just in case?

I manually added tests in the description. Lmk if this is sufficient. 

> Another option is to open another PR to add missing test cases

Sure I can do this, is it okay if I did it in a separate PR after this is merged?

> BTW, what about /var/lib/cni 

This directory stores config files and [results of interaction between](https://github.com/containernetworking/cni/blob/main/SPEC.md) containerd and cni plugins. `/var/lib/cni/networks` is where the config files are stored. For example `/var/lib/cni/networks/bridge/10.4.0.2` contains config for 10.4.0.2 which is the IP given by the CNI plugin to a container.  Each time a container stops are starts it gets a new IP, so may not be too useful to save this info. `/var/lib/cni/results` contains [results](https://github.com/containernetworking/cni/blob/main/SPEC.md#Section-5-Result-Types) when containerd passes parameters to the cni plugin, so maybe its okay to not persist this data.

Reply:
> I manually added tests in the description. Lmk if this is sufficient.

LGTM

> Sure I can do this, is it okay if I did it in a separate PR after this is merged?

Sure.

> This directory stores config files and [results of interaction between](https://github.com/containernetworking/cni/blob/main/SPEC.md) containerd and cni plugins. /var/lib/cni/networks is where the config files are stored. For example /var/lib/cni/networks/bridge/10.4.0.2 contains config for 10.4.0.2 which is the IP given by the CNI plugin to a container. Each time a container stops are starts it gets a new IP, so may not be too useful to save this info. /var/lib/cni/results contains [results](https://github.com/containernetworking/cni/blob/main/SPEC.md#Section-5-Result-Types) when containerd passes parameters to the cni plugin, so maybe its okay to not persist this data.

I see. As long as we are 100% sure nothing there will persist across container-stop-and-start cycle, then we don't have to include that directory.

Reply:
> Maybe we can keep this PR only changing to rootful with still persisting all the directories that were persisted, and then after making sure that everything works fine, we can create a follow-up PR to remove the directories that seem unnecessary?

Sounds good to me. Just making sure that we are not persisting anything under `~/.local` which is not used by rootful mode.

Reply:
One thing to note though is `/var/lib/cni/results` can bloat as described [here](https://github.com/cri-o/cri-o/issues/4296) as a new file is created whenever a container is created.

Reply:
> Just making sure that we are not persisting anything under ~/.local which is not used by rootful mode right?
 
Yes. I should have made it clear that I was referring to the counterpart, not the exact same directory.

Reply:
Thanks! I updated the bind mounts to be consistent with the rootless config.